src/bci_build/package/stunnel.py

```diff
@@ -0,0 +1,48 @@
+"""Application Containers that are generated with the BCI tooling"""
+
+from pathlib import Path
+
+from bci_build.os_version import ALL_NONBASE_OS_VERSIONS
+from bci_build.os_version import CAN_BE_LATEST_OS_VERSION
+from bci_build.package import DOCKERFILE_RUN
+from bci_build.package import ApplicationStackContainer
+from bci_build.package import ParseVersion
+from bci_build.package import Replacement
+from bci_build.package.helpers import generate_from_image_tag
+from bci_build.package.helpers import generate_package_version_check
+from bci_build.registry import publish_registry
+
+STUNNEL_CONTAINERS = [
+    ApplicationStackContainer(
+        name="stunnel",
+        os_version=os_version,
+        tag_version=(tag_ver := "5"),
+        is_latest=os_version in CAN_BE_LATEST_OS_VERSION,
+        from_target_image=generate_from_image_tag(os_version, "bci-micro"),
+        version=(stunnel_version_re := "%%stunnel_re%%"),
+        pretty_name="Stunnel",
+        package_list=["stunnel"],
+        replacements_via_service=[
+            Replacement(stunnel_version_re, package_name="stunnel")
+        ],
+        extra_files={
+            "entrypoint.sh": (
+                (stunnel_dir := Path(__file__).parent / "stunnel") / "entrypoint.sh"
+            ).read_bytes(),
+            "stunnel.conf": (stunnel_dir / "stunnel.conf").read_bytes(),
+        },
+        _publish_registry=publish_registry(os_version, app_collection=True),
+        build_stage_custom_end=generate_package_version_check(
+            "stunnel", tag_ver, ParseVersion.MAJOR, use_target=True
+        ),
+        custom_end=f"""COPY entrypoint.sh /usr/local/bin/
+COPY stunnel.conf /etc/stunnel/stunnel.conf
+{DOCKERFILE_RUN} chmod 0755 /usr/local/bin/entrypoint.sh; \
+    chown --recursive stunnel /etc/stunnel
```

Comment:
shouldn't this just be on the `stunnel.conf` file rather than the directory as well?

Reply:
No, I think it would be better if the `stunnel` user owns the whole directory. The entrypoint will write settings into `/etc/stunnel/conf.d/`, so it makes imho sense for the directory to be owned by the stunnel user. We could resort to just the `conf.d` subdir, but I see little benefit in that (and a lot of potential inconveniences).

Reply:
I don't like directories being owned by a service user as that allows that service user to modify permissions. probably not a big concern here but in general a owner=root,group=servicegroup,mod 0750 or so makes more sense to me. anyhow not a blocking issue.